merge2sorted_list.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+Merge two sorted linked lists and return it as a new list. The new list should be made by splicing together the nodes of the first two lists.
+
+*/
+
+/**
+ * Definition for singly-linked list.
+ * struct ListNode {
+ *     int val;
+ *     ListNode *next;
+ *     ListNode(int x) : val(x), next(NULL) {}
+ * };
+ */
+class Solution {
+public:
+    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
```

Comment:
Consider using "dummyHeadNode" technique for linked list, so you don't have to check NULL for inserting first node.

FYI -- This is not a great link, but a good reference.
 https://jlmedina123.wordpress.com/2014/10/21/linked-list-functions-with-dummy-head-and-without-it/

```
        ListNode dummyNode;
        ListNode *tail = &dummyNode;
        while (l1 && l2) {
            if (l1->val < l2->val) {            
                tail->next = l1;
                l1 = l1->next;
            } else {
                tail->next = l2;
                l2 = l2->next;
            }
            tail = tail->next;
        }
        tail->next = l1 ? l1 : l2;
        return dummyNode.next;
```